app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -164,8 +164,8 @@ def destroy_mapped_tenants
       if source_tenant
         # We just destroyed ourself, reload the source_tenant association
         source_tenant.reload
-        source_tenant.all_subtenants.destroy_all
-        source_tenant.all_subprojects.destroy_all
+        source_tenant.destroy_child_tenants
+        source_tenant.destroy_child_projects
```

Comment:
I feel like a :dependent => :destroy on the relationship would be sufficient for this?